src/download_gcs_file.py

```diff
@@ -0,0 +1,172 @@
+
+    Args:
+        s3_client (boto3.client): S3クライアントオブジェクト。
+        bucket (str): バケット名。
+        prefix (str): オブジェクトのプレフィックス。
+        **kwargs: その他のオプション引数。
+
+    Returns:
+        list: 全てのオブジェクトを含むレスポンスのリスト。
+    """
+    kwargs.update(Bucket=bucket, Prefix=prefix)
+
+    def _list_objects(continuation_token=None):
+        if continuation_token is None:
+            response = s3_client.list_objects_v2(**kwargs)
+        else:
+            response = s3_client.list_objects_v2(
+                ContinuationToken=continuation_token, **kwargs
+            )
+        return response
+
+    responses = []
+    response = _list_objects()
+    responses.append(response)
+    while response["IsTruncated"]:
+        continuation_token = response["NextContinuationToken"]
+        logger.info(f"{continuation_token=}")
+        response = _list_objects(continuation_token)
+        responses.append(response)
+    return responses
+
+
+def _get_negocai_s3_file_list(negocia_profile, negocia_s3_uri):
+    negocia_s3_client = get_s3_client(negocia_profile)
+
+    negocia_s3_uri = Path(negocia_s3_uri)
+    negocia_bucket_name = negocia_s3_uri.parts[1]
+    negocia_s3_dir = "/".join(negocia_s3_uri.parts[2:])
+
+    negocia_response_list = _list_all_objects(
+        negocia_s3_client, negocia_bucket_name, negocia_s3_dir
+    )
+    df_negocia = pd.concat([pd.DataFrame(r["Contents"]) for r in negocia_response_list])
+    return df_negocia.Key.apply(lambda x: Path(x).stem).to_list()
+
+
+def download_file_requester_pays(
+    bucket_name, project_id, source_blob_name, destination_file_name, storage_client
+):
+    bucket = storage_client.bucket(bucket_name, user_project=project_id)
+    blob = bucket.blob(source_blob_name)
+    blob.download_to_filename(destination_file_name)
+
+    print(
+        "Blob {} downloaded to {} using a requester-pays request.".format(
+            source_blob_name, destination_file_name
+        )
+    )
+
+
+def download_worker(args):
+    bucket_name, project_id, source_blob_name, destination_file_name, storage_client = (
+        args
+    )
+    # ダウンロード処理はdownload_file_requester_pays関数をそのまま利用
+    download_file_requester_pays(
+        bucket_name, project_id, source_blob_name, destination_file_name, storage_client
+    )
+
+
+def main():
+    args = create_parser()
+    save_dir = Path(args.save_dir)
+    performance_df = pd.read_csv(args.csv_path)
+    logger.info(f"dowonload file : {len(performance_df.path.unique())}")
+    if args.negocia_s3_uri:
+        target_keys = _get_negocai_s3_file_list(
+            args.negocia_profile, args.negocia_s3_uri
+        )
+        performance_df = performance_df[
+            ~performance_df.id.astype(str).isin(target_keys)
+        ]
+    performance_df = performance_df[["id", "path"]].drop_duplicates()
+    download_dict = dict(zip(performance_df["id"], performance_df["path"]))
+    logger.info(f"dowonload file without s3: {len(download_dict)}")
```

Comment:
- dowonload file -> download file
```suggestion
    logger.info(f"download file without s3: {len(download_dict)}")
```